src/common/cmake.py

```diff
@@ -26,8 +26,7 @@ def configure(self, variant: Variant, build_kit: str = "prod"):
             f" -B{self.workspace_artifacts.get_build_dir(variant, build_kit)}"
             f" -G Ninja "
             f" -DBUILD_KIT:STRING={build_kit}"
-            f" -DFLAVOR:STRING={variant.flavor}"
-            f" -DSUBSYSTEM:STRING={variant.subsystem}"
+            f" -DVARIANT:STRING={variant.to_string()}"

```

Comment:
SPL core needs to know about a variant. The flavor and subsystem information is obsolete.